pkg/operator/operator_test.go

```diff
@@ -158,15 +135,10 @@ func TestOperatorSync_NoOp(t *testing.T) {
 			}
 
 			stopCh := make(<-chan struct{})
-			optr := newFakeOperator(nil, []runtime.Object{infra}, stopCh)
-			optr.imagesFile = imagesFilePath
-
-			err = optr.sync("test-key")
-			if !assert.NoError(t, err, "unexpected sync failure") {
-				t.Fatal()
-			}
+			optr := newFakeOperator(nil, []runtime.Object{newFeatureGate(v1.TechPreviewNoUpgrade), infra}, stopCh)
```

Comment:
Normally, I would be against passing `newFeatureGate(v1.TechPreviewNoUpgrade)` object to the fake informer as it is no needed for the test. Though, the operator processes an item only when a deployment or a feature gate object is created/updated/deleted. Without passing either of those, the operator workers will not process anything. So, passing the feature gate object is actually a trigger to render new deployment. In the real cluster there has to be at least one deployment available in order to trigger the processing loop.